### PR TITLE
feat(type_inference): added NumericDefaultInt cons

### PR DESCRIPTION
**Pull request by Bolladeen**
- Added NumericDefaultInt constraint
 - Updated IntegerLit semantic node to have monotype value
 - Updated semantic analysis to convert all NumericDefaultInt to int in final pass

### Done checklist
- [ ] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: libflux/src/core/semantic/convert.rs [hunk 1]
```diff
@@ -995,6 +996,7 @@ mod tests {
                             value: Expression::Integer(IntegerLit {
                                 loc: b.location.clone(),
                                 value: 10,
+                                typ: MonoType::Int,
```
[the remark this answers is not in the document]
Comment (Bolladeen) [reply libflux/src/core/semantic/convert.rs:999]: Rust seems to complain if I leave on the typ field. Should I just leave it as it or change it to a type variable Monotype?

File: libflux/src/core/semantic/convert.rs [hunk 2]
```diff
@@ -666,13 +667,21 @@ fn convert_float_literal(lit: ast::FloatLit, _: &mut Fresher) -> Result<FloatLit
...
-    Ok(IntegerLit {
+fn convert_integer_literal(lit: ast::IntegerLit, fresher: &mut Fresher) -> Result<PolyNumericLit> {
+    Ok(PolyNumericLit {
         loc: lit.base.location,
         value: lit.value,
+        typ: MonoType::Var(fresher.fresh()),
     })
 }
 
+/*fn convert_integer_literal(lit: ast::IntegerLit, _: &mut Fresher) -> Result<IntegerLit> {
```
Comment (jlapacik) [inline libflux/src/core/semantic/convert.rs:678]: nit: you can remove this

File: libflux/src/core/semantic/tests.rs [hunk 3]
```diff
@@ -153,9 +152,55 @@ fn infer_types(
             );
         }
     }
+
     return Ok(got.into());
 }
 
+/*pub fn convert_numdefault_int(
```
Comment (jlapacik) [inline libflux/src/core/semantic/tests.rs:159]: Can you remove this too?

**Review by jlapacik — COMMENTED** [review]
Just some tiny suggestions but overall this looks very good.

**Pull request closed without merging**